### PR TITLE
{Telemetry|Feedback} add installer data

### LINKED ISSUE #9013: telemetry: include Azure CLI acquisition information in the user-agent
For us to know how people acquired CLI, like pip, bundle, curl? We need this information to help out with the decision that whether and when CLI can be Python3 only

//CC: @marstr

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Resolve #9013

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Telemetry

**Comment by haroldrandom** [comment]
Why added `AZ_INSTALLER`?

File: src/azure-cli-core/azure/cli/core/commands/client_factory.py [hunk 1]
```diff
@@ -186,6 +193,11 @@ def _get_add_headers_callback(cli_ctx):
 
     def _add_headers(request):
         agents = [request.headers['User-Agent'], UA_AGENT]
+        try:
+            agents.append(os.environ[ENV_AZ_INSTALLER])
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/client_factory.py]: The current UA is like 

```
'User-Agent': 'python/3.8.2 (Windows-10-10.0.18362-SP0) msrest/0.6.11 msrest_azure/0.6.2 azure-mgmt-resource/8.0.1 Azure-SDK-For-Python AZURECLI/2.2.0'
```

Maybe we can append the content of `ENV_AZ_INSTALLER` to `UA_AGENT` and make it like `AZURECLI/2.2.0 (PIP)`?

https://github.com/Azure/azure-cli/blob/31a9724478c67751ae0bee6cc0d9b75b763df17c/src/azure-cli-core/azure/cli/core/commands/client_factory.py#L18

This is like what Chrome does: 

```
Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Mobile Safari/537.36
```

My two cents.

File: src/azure-cli-core/azure/cli/core/commands/client_factory.py [hunk 2]
```diff
@@ -83,6 +84,12 @@ def configure_common_settings(cli_ctx, client):
     client.config.enable_http_logger = True
 
     client.config.add_user_agent(UA_AGENT)
+
+    try:
+        client.config.add_user_agent(os.environ[ENV_AZ_INSTALLER])
+    except KeyError:
+        pass
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/client_factory.py]: The same logic is also needed by `az rest`: https://github.com/Azure/azure-cli/blob/31a9724478c67751ae0bee6cc0d9b75b763df17c/src/azure-cli-core/azure/cli/core/util.py#L556-L558

Maybe we can extract the header handling logic to `src/azure-cli-core/azure/cli/core/util.py`?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: #12734 is merged. We can merge from dev and put the logic in `get_az_user_agent`.

File: src/azure-cli-core/azure/cli/core/tests/test_util.py [hunk 3]
```diff
@@ -213,6 +213,11 @@ def test_configured_default_setter(self):
 
     @mock.patch('azure.cli.core.__version__', '7.8.9')
     def test_get_az_user_agent(self):
+        os.environ['AZ_INSTALLER'] = 'PIP'
+        actual = get_az_user_agent()
+        self.assertEqual(actual, 'AZURECLI/7.8.9 (PIP)')
+
+        del os.environ['AZ_INSTALLER']
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/tests/test_util.py]: A cleaner way to do the test: 

```suggestion
        with mock.patch.dict('os.environ', {'AZ_INSTALLER': 'PIP'}):
            actual = get_az_user_agent()
            self.assertEqual(actual, 'AZURECLI/7.8.9 (PIP)')

```

**Pull request merged**